htmresearch/frameworks/pytorch/resnet_models.py

```diff
@@ -0,0 +1,269 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2018, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Modified from torchvision
```

Comment:
It looks like there were minor modifications to `ResNet` class, the rest of the code is identical. Why not just create a class that  inherits `ResNet`? 
You could update the `avgpool.kernel_size` and add the `learningIterations` in the subclass leaving the rest of this code in torchvision.

Reply:
Good point!

Reply:
Just did a quick version of this now, and removed the other classes that were unchanged.